lib/internal/assert/assertion_error.js

```diff
@@ -41,272 +37,77 @@ const kReadableOperator = {
   notDeepEqualUnequal: 'Expected values not to be loosely deep-equal:',
 };
 
-// Comparing short primitives should just show === / !== instead of using the
-// diff.
-const kMaxShortLength = 12;
-
 function copyError(source) {
-  const target = ObjectAssign({ __proto__: ObjectGetPrototypeOf(source) }, source);
-  ObjectDefineProperty(target, 'message', { __proto__: null, value: source.message });
+  const target = ObjectAssign(
+    { __proto__: ObjectGetPrototypeOf(source) },
+    source,
+  );
+  ObjectDefineProperty(target, 'message', {
+    __proto__: null,
+    value: source.message,
+  });
   return target;
 }
 
 function inspectValue(val) {
   // The util.inspect default values could be changed. This makes sure the
   // error messages contain the necessary information nevertheless.
-  return inspect(
-    val,
-    {
-      compact: false,
-      customInspect: false,
-      depth: 1000,
-      maxArrayLength: Infinity,
-      // Assert compares only enumerable properties (with a few exceptions).
-      showHidden: false,
-      // Assert does not detect proxies currently.
-      showProxy: false,
-      sorted: true,
-      // Inspect getters as we also check them when comparing entries.
-      getters: true,
-    },
-  );
+  return inspect(val, {
+    compact: false,
+    customInspect: false,
+    depth: 1000,
+    maxArrayLength: Infinity,
+    // Assert compares only enumerable properties (with a few exceptions).
+    showHidden: false,
+    // Assert does not detect proxies currently.
+    showProxy: false,
+    sorted: true,
+    // Inspect getters as we also check them when comparing entries.
+    getters: true,
+  });
 }
 
 function getErrorMessage(operator, message) {
   return message || kReadableOperator[operator];
 }
 
-function createErrDiff(actual, expected, operator, message = '') {
-  let other = '';
-  let res = '';
-  let end = '';
-  let skipped = false;
-  const actualInspected = inspectValue(actual);
-  const actualLines = StringPrototypeSplit(actualInspected, '\n');
-  const expectedLines = StringPrototypeSplit(inspectValue(expected), '\n');
+function showSimpleDiff(test, message = '') {
+  const isPrimitive = test !== Object(test);
+  const isEmptyArray = ArrayIsArray(test) && test.length === 0;
+  const isSimpleObject = !isPrimitive && !(test instanceof Map) && ObjectKeys(test).length === 0;
```

Comment:
```suggestion
  if (typeof test !== 'object' || test === null) {
    return true;
  }
  const maxCharactersToInspect = Math.min(util.inspect.defaults.breakLength + 10, 1000)
  return inspect(test).slice(0, maxCharactersToInspect).includes('\n')
```

That way it is a) faster (`Object.keys()` is expensive for big objects) and b) it is going to match anything that is only a single line long, not only things we thought about up front.